File: connectors/connector-redmine/src/main/scala/com/taskadapter/connector/redmine/CustomFieldDefinitionFinder.scala
package com.taskadapter.connector.redmine

import java.util

import com.taskadapter.model.Field
import com.taskadapter.redmineapi.bean.CustomFieldDefinition

import scala.collection.JavaConverters._

object CustomFieldDefinitionFinder {
  def findCustomFieldId(customFieldDefinitions: util.List[CustomFieldDefinition], field: Field[_]) : Integer = {
    customFieldDefinitions.asScala.find(d => d.getName == field.name).map(_.getId).orNull
  }
}

File: connectors/connector-redmine/src/main/java/com/taskadapter/connector/redmine/CustomFieldDefinitionFinder.java
package com.taskadapter.connector.redmine;

import com.taskadapter.model.Field;
import com.taskadapter.redmineapi.bean.CustomFieldDefinition;

import java.util.List;

public class CustomFieldDefinitionFinder {
    public static Integer findCustomFieldId(List<CustomFieldDefinition> customFieldDefinitions, Field<?> field) {
        return customFieldDefinitions.stream()
                .filter(d -> d.getName().equals(field.getFieldName()))
                .findFirst().map(CustomFieldDefinition::getId)
                .orElse(null);
    }
}
